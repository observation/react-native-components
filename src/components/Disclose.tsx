import React from 'react'
import { StyleSheet, TouchableOpacity, View, Text, ViewStyle, TextStyle, StyleProp } from 'react-native'

import { Icon } from './Icon'
import Log from '../lib/Log'
import appTextStyle from '../styles/text'
import theme from '../styles/theme'

type Props = {
  text: string
  onPress?: () => void
  containerStyle?: StyleProp<ViewStyle>
  textStyle?: StyleProp<TextStyle>
}

const Disclose = ({ text, onPress, textStyle, containerStyle }: Props) => {
  Log.debug('Disclose')
  return (
    <TouchableOpacity activeOpacity={0.5} onPress={onPress}>
      <View style={[styles.containerStyle, containerStyle]}>
        <Text style={[appTextStyle.link, textStyle]}>{text}</Text>
        <Icon name="chevron-right" size={theme.icon.size.extraLarge} />
      </View>
    </TouchableOpacity>
  )
}

export default Disclose

const styles = StyleSheet.create({
  containerStyle: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    marginHorizontal: theme.margin.common,
    backgroundColor: theme.color.white,
  },
})
