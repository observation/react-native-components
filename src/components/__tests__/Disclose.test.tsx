import React from 'react'

import { fireEvent, render } from '@testing-library/react-native'

import Disclose from '../Disclose'

const onPress = jest.fn()

describe('Disclose', () => {
  describe('Rendering', () => {
    test('Only text', () => {
      // WHEN
      const { toJSON } = render(<Disclose text="some text" />)

      // THEN
      expect(toJSON()).toMatchSnapshot()
    })

    test('Styles and callback', () => {
      // GIVEN
      const { toJSON } = render(
        <Disclose text="some text" containerStyle={{ flex: 1 }} textStyle={{ fontSize: 1 }} onPress={onPress} />,
      )

      // THEN
      expect(toJSON()).toMatchSnapshot()
    })
  })

  test('Interaction', () => {
    // GIVEN
    const { getByText } = render(<Disclose text="some text" onPress={onPress} />)

    // WHEN
    fireEvent.press(getByText(/some text/))

    // THEN
    expect(onPress).toBeCalled()
  })
})
