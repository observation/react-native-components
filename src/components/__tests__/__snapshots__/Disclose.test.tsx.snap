// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Disclose Rendering Only text 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginHorizontal": 16,
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#0066B1",
            "fontFamily": "Ubuntu",
            "fontSize": 14,
            "fontStyle": "normal",
            "fontWeight": "normal",
            "lineHeight": 20,
          },
          undefined,
        ]
      }
    >
      some text
    </Text>
    <Icon
      color="#0066B1"
      name="chevron-right"
      size={18}
      type="light"
    />
  </View>
</View>
`;

exports[`Disclose Rendering Styles and callback 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginHorizontal": 16,
        },
        {
          "flex": 1,
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#0066B1",
            "fontFamily": "Ubuntu",
            "fontSize": 14,
            "fontStyle": "normal",
            "fontWeight": "normal",
            "lineHeight": 20,
          },
          {
            "fontSize": 1,
          },
        ]
      }
    >
      some text
    </Text>
    <Icon
      color="#0066B1"
      name="chevron-right"
      size={18}
      type="light"
    />
  </View>
</View>
`;
